boards/px4/fmu-v3/src/init.c

```diff
@@ -198,19 +199,18 @@ __EXPORT void board_on_reset(int status)
  *
  ************************************************************************************/
 
-static int determin_hw_version(int *version, int *revision)
+static int determine_hw_version(void)
```

Comment:
The API with the params `int *version`, `int *revision` is compatible with the real HW revision and version. Why change it?

Reply:
I looked at the function declaration and two things caught my eye:
- The revision was never changed, it's always 0, clearly it's not used.
- To be pedantic, the function is called determine hw "version" (and not revision).
- The return value of the function was always 0/OK so it's a bit of a lie that it would tell you when it fails. And the caller also does not actually deal with the return value not being 0/OK which I didn't like either.

Reply:
All good reason.

In my mind a Version is a type of thing, Revision is how many times it to to get right. Both can have changes that effect HW.  I prefer having one system level interface that supports both, even though we could not support the Revision on the pre exiting HW.  

In retrospect and thinking about a split for FMUM/BASE the name `determin_hw_version`  could be better named as `determin_hw_characteristics` and  2 values could then be interpreted as needed.  Or wrapped in a union of structs with names.

Reply:
Ok, I can see that there is actually the function `determine_hw_info` in quite a few places which actually has the arguments swapped. I suppose I should change it to that API?

Reply:
Yes - I think it would be good to be consistent.

Reply:
Ok, pushed.